### DISCUSSION TITLE
[Question] How does DocFX handle reading xml documentation files?

### DISCUSSION THREAD

**Discussion by wouterroos** [dotnet/docfx#8273] [General]
Hi,

I'm in the somewhat unique position that I'd like to alter the xml documentation file of a given assembly before instructing DocFX to generate a website from it. I'm wondering how DocFX handles this exactly since it doesn't seem to use the regular build output from a csproj file. 

Any insights would be greatly appreciated!

Thanks,

**Comment by KalleOlaviNiemitalo** [comment]
If you list solution files or project files in the [metadata section](https://dotnet.github.io/docfx/tutorial/docfx.exe_user_manual.html#31-properties-for-metadata) of docfx.json, then DocFX reads XML documentation comments from the corresponding source files via the Roslyn API. It might be possible to alter the documentation with either Roslyn plugins or DocFX plugins but I don't know the details.

If you instead list assembly files (`.dll`, `.exe`) in docfx.json, then DocFX reads XML documentation from similarly named `.xml` files. Thus, you can alter those files after you run the compiler and before you run DocFX.

[Overwrite files](https://dotnet.github.io/docfx/tutorial/intro_overwrite_files.html) might be another alternative.
